include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -971,7 +971,7 @@ namespace VectorTools
       ReductionControl     control(rhs.size(), 0., 1e-12, false, false);
       SolverCG<LinearAlgebra::distributed::Vector<Number> > cg(control);
       typename PreconditionChebyshev<MatrixType, LocalVectorType>::AdditionalData data;
-      data.matrix_diagonal_inverse = mass_matrix.get_matrix_diagonal_inverse();
+      data.matrix_diagonal_inverse = mass_matrix.get_matrix_diagonal_inverse().get_vector();
```

Comment:
@kronbichler If I am not mistaken, there is a better way of using `DiagonalMatrix` in `PreconditionChebyshev`.